webextension/catcher.js

```diff
@@ -1,22 +1,38 @@
-/* globals callBackground */
-
 window.catcher = (function () {
   let exports = {};
 
-  /** Call with an error object (with .name, .message, .help, etc) */
-  exports.unhandled = function unhandled(error, info) {
-    callBackground("reportError", makeError(error, info));
+  let handler;
+
+  let queue = [];
```

Comment:
Ideally nothing earlier in the load process than the registered handler will emit an error – but if it does, we won't get Sentry notification without this queue, and we might miss errors.  The queue ensures that very early errors won't get dropped.